src/app/view/OsReceive/useOsReceiveApi.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect } from 'react'
+
+import { useNativeIntent } from 'cozy-intent'
+
+import { useOsReceiveState } from '/app/view/OsReceive/OsReceiveState'
+import { OsReceiveLogger } from '/app/domain/osReceive'
+
+export const useOsReceiveApi = (): void => {
+  const state = useOsReceiveState()
+  const nativeIntent = useNativeIntent()
+
+  useEffect(() => {
+    if (state.filesUploaded.length === 0 || !nativeIntent) return
+    const isLast = state.filesUploaded.length === state.filesToUpload.length
+
+    const onUploaded = async (): Promise<void> => {
+      OsReceiveLogger.info('onUploaded called')
+
+      try {
+        const res = await nativeIntent.call(
+          'drive.dev.192-168-1-65.nip.io',
```

Comment:
TODO: you have hard coded value here

Reply:
replaced by using routeToUpload.href value (that formatting is valid for cozy-intent when retrieving the messenger)
```
  const res = await nativeIntent.call(
    state.routeToUpload.href,
    'onFileUploaded',
    state.filesUploaded[state.filesUploaded.length - 1],
    isLast ? state.filesUploaded.length : undefined
  )
```